### PR TITLE
Handle no-support for getTransports and getClientExtensionResults on some browsers.

### PR DESCRIPTION
**Pull request by chaitanyaachanta-okta**
## Description:

- As part of this [change](https://github.com/okta/okta-signin-widget/pull/2914), we added support to send additional info to backend after webauthn enrollment. 
But this caused regression due to non-support of these APIs on few browsers https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential/getClientExtensionResults 
https://developer.mozilla.org/en-US/docs/Web/API/AuthenticatorAttestationResponse/getTransports
- Fix is to check whether these APIs are supported before calling them.

## PR Checklist

- [X] Have you verified the basic functionality for this change?
- [X] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-571393](https://oktainc.atlassian.net/browse/OKTA-571393)

### Reviewers:
@okta/enterpriseauth-team 

## OIE Videos

CHROME

https://user-images.githubusercontent.com/100623681/215632873-946d3363-31c1-4118-b59a-42b510f23ffd.mov

FIREFOX

https://user-images.githubusercontent.com/100623681/215632912-39673ccc-6abe-4196-a94c-a0ac5821172b.mov

## V1 VIDEOS

CHROME

https://user-images.githubusercontent.com/100623681/215634393-f11aee4d-89a1-4939-8e5e-7f43dd61c086.mov

FIREFOX

https://user-images.githubusercontent.com/100623681/215634445-2033b14c-b692-4cf5-8373-332ec2e55711.mov

### Downstream Monolith Build:

https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&page=1&pageSize=6&sha=60881b43c2f529bcb8b21adbe9c4cbc9e093fd02&tab=main

### CODE REVIEW

File: src/util/webauthn.js [hunk 1]
```diff
@@ -52,9 +52,17 @@ function getAssertion(challenge, allowList) {
   return adaptToOkta(promise);
 }
 
+function processWebAuthnResponse(inputFunction) {
+  if (typeof inputFunction === 'undefined' || inputFunction() === undefined || inputFunction() === null) {
```
Comment (naoitoi-okta) [inline src/util/webauthn.js:56]: A couple of Internet posts suggest this as a check to see if a value represents a function.  It probably does not matter, but FYI.

if (typeof test === 'function') {

https://www.tutorialspoint.com/How-to-check-if-a-JavaScript-function-is-defined#:~:text=Using%20the%20typeof%20Operator,whether%20the%20function%20is%20defined.
https://stackoverflow.com/questions/1042138/how-to-check-if-function-exists-in-javascript

**Review by naoitoi-okta — DISMISSED** [review]

File: src/util/webauthn.js (hunk 1 above)
Comment (praveendurai-okta) [reply src/util/webauthn.js:56]: We are calling the `inputFunction` multiple times here. Instead can we call it only once and check IF condition based on the response?

**Review by mpeng-okta — DISMISSED** [review]

**Review by praveendurai-okta — DISMISSED** [review]
Just one comment. Rest looks good to me.

**Review by jaredperreault-okta — APPROVED** [review]

**Review by glenfannin-okta — APPROVED** [review]

**Review by naoitoi-okta — APPROVED** [review]

**Comment by chaitanyaachanta-okta** [comment]
Sorry, as we need to do TF. All changes needs to be merged to 7.2 not master. So created another PR for that.
https://github.com/okta/okta-signin-widget/pull/3058
Closing this now.

**Pull request closed without merging**